src/whitenoise/groups.rs

```diff
@@ -502,72 +498,247 @@ impl Whitenoise {
+    /// * `image_nonce` - Encryption nonce
+    ///
+    /// # Returns
+    /// Path to the cached image file
+    async fn download_and_cache_group_image(
+        &self,
+        account_pubkey: &PublicKey,
+        group_id: &GroupId,
+        image_hash: &[u8; 32],
+        image_key: &[u8; 32],
+        image_nonce: &[u8; 12],
+    ) -> Result<PathBuf> {
+        // Check if already cached (look for any file with this hash prefix)
+        let hash_hex = hex::encode(image_hash);
+        let media_files = self.media_files();
+        if let Some(cached_path) =
+            media_files.find_file_with_prefix(group_id, "group_images", &hash_hex)
+        {
+            tracing::debug!(
+                target: "whitenoise::groups::download_and_cache_group_image",
+                "Group image already cached at: {}",
+                cached_path.display()
+            );
+
+            // Update accessed_at in database
+            use crate::whitenoise::media_files::MediaFileUpload;
+            let upload = MediaFileUpload {
+                data: &[],
+                file_hash: *image_hash,
+                mime_type: "", // MIME type not needed for update
+                media_type: "group_image",
+                blossom_url: None,
+                dimensions: None,
+                blurhash: None,
+            };
+            if let Err(e) = media_files
+                .record_in_database(account_pubkey, group_id, &cached_path, upload)
+                .await
+            {
+                tracing::warn!(
+                    target: "whitenoise::groups::download_and_cache_group_image",
+                    "Failed to update accessed_at in database: {}",
+                    e
+                );
+            }
+
+            return Ok(cached_path);
+        }
+
+        tracing::info!(
+            target: "whitenoise::groups::download_and_cache_group_image",
+            "Downloading group image {} for group {}",
+            hex::encode(image_hash),
+            hex::encode(group_id.as_slice())
+        );
+
+        // Download encrypted blob from Blossom
+        let blossom_server = "https://blossom.primal.net";
+        let blossom_url = reqwest::Url::parse(&format!("{}/{}", blossom_server, hash_hex))
+            .map_err(|e| WhitenoiseError::Other(anyhow::anyhow!("Invalid Blossom URL: {}", e)))?;
+
+        let client = reqwest::Client::new();
+        let response = client.get(blossom_url.as_str()).send().await.map_err(|e| {
+            WhitenoiseError::BlossomDownload(format!("Failed to download image: {}", e))
+        })?;
+
+        if !response.status().is_success() {
+            return Err(WhitenoiseError::BlossomDownload(format!(
+                "Download failed with status: {}",
+                response.status()
+            )));
+        }
+
+        let encrypted_data = response.bytes().await.map_err(|e| {
+            WhitenoiseError::BlossomDownload(format!("Failed to read response: {}", e))
+        })?;
```

Comment:
Maybe we do need a custom whitenoise BlossomClient "layer" to do this, it feels very out of place here.